File: sourcePY/287.py
    def findDuplicate(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        slow, fast = 0, 0
        while True:
            slow = nums[slow]
            fast = nums[nums[fast]]
            if slow == fast:
                break
        find = 0
        while find != slow:
            find = nums[find]
            slow = nums[slow]
        return find

File: source/287.cpp
	int findDuplicate(vector<int>& nums) {
		int l = 0, r = nums.size() - 1, mid = 0;
		while (l <= r) {
			mid = l + (r - l) / 2;
			int count = 0;
			for (int i = 0; i < nums.size(); ++i) {
				if (nums[i] <= mid) ++count;
			}
			if (count > mid) r = mid - 1;
			else l = mid + 1;
		}
		return l;
	}